extension/src/frequency.rs

```diff
@@ -233,6 +233,47 @@ impl SpaceSavingTransState {
         }
     }
 
+    fn ingest_aggregate_data(
+        &mut self,
+        val_count: u64,
+        values: &DatumStore,
+        counts: &[u64],
+        overcounts: &[u64],
+    ) {
+        self.total_vals = val_count;
+
+        for (idx, datum) in values.iter().enumerate() {
+            self.entries.push(SpaceSavingEntry {
```

Comment:
Should these assert entries is empty?

It seems these are just constructor part two; have you considered folding the calls in?

Reply:
Asserting entries is empty is probably a good idea.  As for constructors, the problem is that there are a couple constructors and which one is used is completely independent of which call we use to fill them.  So if I wanted to fold them in I'd have to have a function for every combination of calls.